src/EnergyPlus/FanCoilUnits.cc

```diff
@@ -503,44 +505,40 @@ namespace FanCoilUnits {
                 }
             }
 
-            errFlag = false;
-            ValidateComponent(state, fanCoil.FanType, fanCoil.FanName, errFlag, CurrentModuleObject);
-            if (errFlag) {
-                ShowContinueError(state, format("specified in {} = \"{}\".", CurrentModuleObject, fanCoil.Name));
+            fanCoil.fanType = static_cast<DataHVACGlobals::FanType>(getEnumValue(DataHVACGlobals::fanTypeNamesUC, Alphas(9)));
+            if (fanCoil.fanType == DataHVACGlobals::FanType::Invalid) {
+                ShowSevereInvalidKey(state, eoh, cAlphaFields(9), Alphas(9));
                 ErrorsFound = true;
-            } else {
-                if (!Util::SameString(fanCoil.FanType, "Fan:SystemModel")) {
-                    Fans::GetFanType(state, fanCoil.FanName, fanCoil.FanType_Num, errFlag, CurrentModuleObject, fanCoil.Name);
-                    // need to grab fan index here
-                    Fans::GetFanIndex(state, fanCoil.FanName, fanCoil.FanIndex, errFlag, fanCoil.FanType);
-                    fanCoil.fanAvailSchIndex = Fans::GetFanAvailSchPtr(state, fanCoil.FanType, fanCoil.FanName, errFlag);
-                    if (errFlag) {
-                        ShowContinueError(state, format("Occurs in {} = {}", CurrentModuleObject, fanCoil.Name));
-                        ErrorsFound = true;
-                        errFlag = false;
-                    }
-                    switch (fanCoil.FanType_Num) {
-                    case DataHVACGlobals::FanType_SimpleConstVolume:
-                    case DataHVACGlobals::FanType_SimpleVAV:
-                    case DataHVACGlobals::FanType_SimpleOnOff: {
+            } else if (fanCoil.fanType != DataHVACGlobals::FanType::SystemModel) {
+                fanCoil.FanIndex = Fans::GetFanIndex(state, fanCoil.FanName);
+                if (fanCoil.FanIndex == 0) {
+                    ShowSevereItemNotFound(state, eoh, cAlphaFields(10), fanCoil.FanName);
+                    ErrorsFound = true;    
+                } else {
+                    assert(fanCoil.fanType == Fans::GetFanType(state, fanCoil.FanIndex));
```

Comment:
If a user selects the wrong fan object type in a fan coil unit there would be node connection warnings in release but no specific warning from the fan coil. 

FanCoilAutoSize.idf:

    ZoneHVAC:FourPipeFanCoil,
      Fan:OnOff,      !- Supply Air Fan Object Type
      Zone1FanCoilFan,         !- Supply Air Fan Name

    Fan:ConstantVolume,
      Zone1FanCoilFan,         !- Name

    ** Warning ** Node Connection Error for object FAN:CONSTANTVOLUME=ZONE1FANCOILFAN
    **   ~~~   **   Air Nodes not on any Branch or Parent Object
    **   ~~~   **   Inlet Node : ZONE1FANCOILOAMIXEROUTLETNODE
    **   ~~~   **   Outlet Node: ZONE1FANCOILFANOUTLETNODE
    ** Warning ** Node Connection Error for object FAN:ONOFF=ZONE1FANCOILFAN
    **   ~~~   **   UNDEFINED not on any Branch or Parent Object
    **   ~~~   **   Inlet Node : ZONE1FANCOILOAMIXEROUTLETNODE
    **   ~~~   **   Outlet Node: UNDEFINED

In debug this assert will fail. I would suggest changing this to a severe warning. Using either this or a more detailed warning (probably the latter).

    if (fanCoil.fanType != Fans::GetFanType(state, fanCoil.FanIndex)) {
        ShowSevereItemNotFound(state, eoh, cAlphaFields(9), fanCoil.FanName);
        ShowSevereItemNotFound(state, eoh, cAlphaFields(9), Alphas(9));
        ErrorsFound = true;
    }

    ** Severe  ** GetFanCoilUnits: ZoneHVAC:FourPipeFanCoil = ZONE1FANCOIL
    **   ~~~   ** Supply Air Fan Name = ZONE1FANCOILFAN, item not found.
    ** Severe  ** GetFanCoilUnits: ZoneHVAC:FourPipeFanCoil = ZONE1FANCOIL
    **   ~~~   ** Supply Air Fan Object Type = FAN:ONOFF, item not found.
    **  Fatal  ** GetFanCoilUnits: Errors found in input. Preceding condition(s) cause termination.